In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis as LDA
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

In [2]:
used_cars = pd.read_csv('used_car_sale_prices.csv')

In [9]:
for col in used_cars.select_dtypes(include=['object']).columns:
    print(f'{col}: {used_cars[col].nunique()}')

Model: 319
Fuel_Type: 3
Color: 17
Automatic: 3
Mfr_Guarantee: 2
BOVAG_Guarantee: 2
ABS: 2
Airbag_1: 2
Airbag_2: 2
Airco: 2
Automatic_airco: 2
Boardcomputer: 2
CD_Player: 2
Central_Lock: 2
Powered_Windows: 2
Power_Steering: 2
Radio: 2
Mistlamps: 2
Sport_Model: 2
Backseat_Divider: 2
Metallic_Rim: 2
Radio_cassette: 2
Parking_Assistant: 2
Tow_Bar: 2
Mfg_Date: 77


## Categorical (Uncleaned)

- Model: 319
- Fuel_Type: 3
- Color: 17
- Automatic: 3
- Mfr_Guarantee: 2
- BOVAG_Guarantee: 2
- ABS: 2
- Airbag_1: 2
- Airbag_2: 2
- Airco: 2
- Automatic_airco: 2
- Boardcomputer: 2
- CD_Player: 2
- Central_Lock: 2
- Powered_Windows: 2
- Power_Steering: 2
- Radio: 2
- Mistlamps: 2
- Sport_Model: 2
- Backseat_Divider: 2
- Metallic_Rim: 2
- Radio_cassette: 2
- Parking_Assistant: 2
- Tow_Bar: 2
- Mfg_Date: 77